x/slashing/types/events.go

```diff
@@ -3,16 +3,14 @@ package types
 // Slashing module event types
 const (
 	EventTypeSlash    = "slash"
+	EventTypeJailed   = "jail"
 	EventTypeLiveness = "liveness"
 
-	AttributeKeyAddress      = "address"
-	AttributeKeyHeight       = "height"
-	AttributeKeyPower        = "power"
-	AttributeKeyReason       = "reason"
```

Comment:
I would prefer to keep reason to have more granularity into the slash and why it occurred.

Reply:
Please undo these changes. The reason event is a must. Consider the case where a client wants to get all slash events but doesn't have any particular validator in mind.

Reply:
Would it be more preferred to simply add another event akin to `CoinsSlashed` in the staking modules's slash function that would just emit how many coins were slashed from the validator addr? Or perhaps this another problem that could be solved by ADR 038? @marbar3778

Reply:
This isn't solved by ADR-038. And adding coins doesn't solve it either. You need a static event value, e.g. `reason`.

Reply:
Adding the amount of coins slashed to the event is the main idea of the PR, see the referenced [issue](https://github.com/cosmos/cosmos-sdk/issues/9138). It wasn't to just trash the reason field - that was just an unfortunate side effect of the route taken to get coins into the event. 

What I was proposing in my last comment was to revert the current changes in this PR and add a new emitted event to the staking keeper's slash function which includes the amount slashed. The downside here is that it would result in 2 events per slash - the ones that are already emitted as they are in master, and a new event emission with the `amount_slashed` information.

I have also thought of a few other ways to get the amount of coins into the event, however they include method signature changes so not sure if ideal:
- update the slash function to return the amount of coins slashed
or
- accept a `reason` as an argument to `slash` and emit there (slashing is emitted in 2 different places currently, this would put it in only one function).

I referenced ADR-038 as marko had recently expressed feelings against emitting more events in today's arch call, and in this [issue](https://github.com/cosmos/cosmos-sdk/issues/9375#issuecomment-854477756), so was curious if it would solve the issue of the slash event not emitting the amount slashed instead of the changes proposed above.

thoughts? @alexanderbez @marbar3778

Reply:
I feel like keeping the slashing events in the x/slashing module is preferable. So it seems to me, returning the amount slashed is the path of least resistance.

Reply:
I've [implemented](https://github.com/cosmos/cosmos-sdk/pull/9458/commits/988bc482930b5ab97f20308dd68a0c759924ceaa) returning the amount slashed - lmk if this looks ok!